src/app/common-components/bread-crumbs/bread-crumbs.component.ts

```diff
@@ -6,8 +6,8 @@ import { Component, Input, OnChanges } from "@angular/core";
   styleUrls: ["./bread-crumbs.component.less"]
 })
 export class BreadCrumbsComponent implements OnChanges {
-  @Input() path = [];
-  private completePath = "";
+  @Input() public path = [];
+  public completePath = "";
 
   ngOnChanges() {
     this.completePath = this.path.join("/ ");
```

Comment:
You see, as far as BreadCrumbsComponent does not use onPush strategy, ngOnChanges can be called every change detection cycle.
To avoid it, there are several ways:
1.) Using a pipe to join path. Pipe can be used inside BreadCrumbsComponent or in the parent component on property binding.
2.) `@Input() public set path(path: Array<string>) {`
`this.completePath= path.join("/ ");`
`}`

`public get path(): string {`
`return this.completePath;`
`}`
where completePath is private property